### PR TITLE
Reduce microbatch size

### LINKED ISSUE #678: Changing microbatch size to 1
Since many people face memory issues in practice, should we change the microbatch size in all scripts from 4 to 1? 

- This won't affect the model's loss and predictive performance due to gradient accumulation
- There is virtually no difference in terms of speed as we can tell based on the new resource tables

I think this change will help with lowering the barrier to entry (remove hurdles) when using lit-gpt

### PR DESCRIPTION
**Pull request by rasbt**
I was going through old issues, and we discussed lowering the micro batch size to 1 a while back in #678 to make the finetuning defaults more compatible with more models and GPUs out there. This will of course increase the runtime, but given that VRAM is usually the bigger bottleneck, I think it's a worthwhile change. Especially now that we have the config files in the `config_hub`, we also have good recommendations for hparams that work well for specific models so I think lowering resource needs is maybe preferred here.

Fixes #678

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]
LGTM

Are the rest of the defaults still good for the "stablelm-base-alpha-3b" model? 

I think we haven't checked in a very long time that the defaults without running a config file leads to meaningful results. It might be easiest if we simply pick one of our configs as the canonical example and update the defaults here accordingly. Just as a suggestion for a follow up if we feel like it.

**Comment by rasbt** [comment]
I agree and updated everything accordingly.

File: litgpt/utils.py [hunk 1]
```diff
@@ -305,7 +305,7 @@ def get_default_supported_precision(training: bool) -> str:
 
     if MPSAccelerator.is_available() or (torch.cuda.is_available() and not torch.cuda.is_bf16_supported()):
         return "16-mixed" if training else "16-true"
-    return "bf16-mixed" if training else "bf16-true"
```
[the remark this answers is not in the document]
Comment (rasbt) [reply litgpt/utils.py:308]: Ah yes, I maybe should have discussed this. The issue here is that it will be very very slow and resource intensive by default. But I am happy to not do this change.

**Comment by rasbt** [comment]
> @rasbt My suggestion was meant as a follow up. The PR was totally fine as it was.

haha ok, in this case I am happy to revert.
